python/cudf/cudf/core/column/datetime.py

```diff
@@ -329,9 +329,18 @@ def infer_format(element, **kwargs):
         raise ValueError("Unable to infer the timestamp format from the data")
 
     if len(second_part) > 1:
-        second_part = pd.core.tools.datetimes._guess_datetime_format(
-            "".join(second_part[1:]), **kwargs
-        )
+        # "Z" indicates Zulu time(widely used in aviation) - Which is
+        # UTC timezone that currently cudf only supports. Having any other
+        # unsuppported timezone will let the code fail below
+        # with a ValueError.
+        second_part.remove("Z")
+        second_part = "".join(second_part[1:])
+
+        if len(second_part) > 1:
+            # Only infer if second_part is not an empty string.
+            second_part = pd.core.tools.datetimes._guess_datetime_format(
+                "".join(second_part[1:]), **kwargs
```

Comment:
The logic is a bit difficult to follow here -- we are doing `second_part[1:]` above, and then `second_part[1:]` again here. Is this intended?

Reply:
Ah, this was unintentional. Thanks for catching this. Fixed it 👍